code/modules/mapping/space_management/multiz_helpers.dm

```diff
@@ -28,24 +29,30 @@
 		return (dir | get_dir(us, them))
 
 /turf/proc/above()
+	if(turf_flags & RESERVATION_TURF)
+		var/datum/turf_reservation/map_reserve = SSmapping.get_reservation_from_turf(src)
+		return map_reserve.get_turf_above(src)
 	return GET_TURF_ABOVE(src)
 
 /turf/proc/below()
+	if(turf_flags & RESERVATION_TURF)
+		var/datum/turf_reservation/map_reserve = SSmapping.get_reservation_from_turf(src)
+		return map_reserve.get_turf_below(src)
 	return GET_TURF_BELOW(src)
```

Comment:
bro this proc is a helper, add this to the macro don't just revert all uses of it. It's like that because it's very hot in multiz and I want to avoid proc overhead